File: baba-chatbot-net/src/Baba.Chatbot.Integrations/Llm/PromptRepository.cs
using Microsoft.Extensions.Configuration;
using Microsoft.Extensions.Logging;

namespace Baba.Chatbot.Integrations.Llm;

public class PromptRepository
{
    private readonly ILogger<PromptRepository> _logger;
    private readonly string _promptsPath;

    public PromptRepository(IConfiguration configuration, ILogger<PromptRepository> logger)
    {
        _logger = logger;
        _promptsPath = configuration["Rag:PromptsPath"] ?? "./config/prompts";
    }

    public async Task<string> GetSystemPromptAsync()
    {
        return await LoadPromptAsync("system.md");
    }

    public async Task<string> GetGuardrailsPromptAsync()
    {
        return await LoadPromptAsync("guardrails.md");
    }

    public async Task<string> GetValuePropPromptAsync()
    {
        return await LoadPromptAsync("value-prop.md");
    }

    public async Task<string> GetResponseStylePromptAsync()
    {
        return await LoadPromptAsync("response-style.md");
    }

    private async Task<string> LoadPromptAsync(string fileName)
    {
        try
        {
            var filePath = Path.Combine(_promptsPath, fileName);
            
            if (!File.Exists(filePath))
            {
                _logger.LogWarning("Prompt file not found: {FilePath}", filePath);
                return string.Empty;
            }

            return await File.ReadAllTextAsync(filePath);
        }
        catch (Exception ex)
        {
            _logger.LogError(ex, "Error loading prompt file: {FileName}", fileName);
            return string.Empty;
        }
    }
}

File: baba-chatbot-python/src/baba_chatbot/integrations/llm/prompt_repository.py
"""
Repositorio de prompts del sistema.
Carga y gestiona los prompts desde archivos markdown.
"""
import os
import logging
from pathlib import Path


logger = logging.getLogger(__name__)


class PromptRepository:
    """
    Repositorio para cargar y gestionar prompts del sistema.
    """

    def __init__(self, prompts_path: str = None):
        """
        Inicializa el repositorio de prompts.
        
        Args:
            prompts_path: Ruta al directorio de prompts
        """
        if prompts_path is None:
            prompts_path = os.path.join(os.getcwd(), "config", "prompts")
        
        self._prompts_path = prompts_path

    async def get_system_prompt(self) -> str:
        """
        Obtiene el prompt del sistema.
        
        Returns:
            Contenido del prompt del sistema
        """
        return await self._load_prompt("system.md")

    async def get_guardrails_prompt(self) -> str:
        """
        Obtiene el prompt de guardrails.
        
        Returns:
            Contenido del prompt de guardrails
        """
        return await self._load_prompt("guardrails.md")

    async def get_value_prop_prompt(self) -> str:
        """
        Obtiene el prompt de propuesta de valor.
        
        Returns:
            Contenido del prompt de propuesta de valor
        """
        return await self._load_prompt("value-prop.md")

    async def get_response_style_prompt(self) -> str:
        """
        Obtiene el prompt de estilo de respuestas.
        
        Returns:
            Contenido del prompt de estilo de respuestas
        """
        return await self._load_prompt("response-style.md")

    async def _load_prompt(self, file_name: str) -> str:
        """
        Carga un archivo de prompt específico.
        
        Args:
            file_name: Nombre del archivo de prompt
            
        Returns:
            Contenido del prompt o cadena vacía si hay error
        """
        try:
            file_path = os.path.join(self._prompts_path, file_name)
            
            if not os.path.exists(file_path):
                logger.warning(f"Prompt file not found: {file_path}")
                return ""

            with open(file_path, 'r', encoding='utf-8') as f:
                content = f.read()
                return content
        except Exception as ex:
            logger.error(f"Error loading prompt file {file_name}: {ex}")
            return ""

